[Ang Yuze] iP

YZmunchmunch opened the pull request:
# **MUNCH**

> "Eat your sorrows away" - nobody ( [source](https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/cb647821-dc91-4178-847f-42c98fbcd407/deogqv8-c614049b-bcf6-440a-9765-9f0b8299e350.png?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7InBhdGgiOiJcL2ZcL2NiNjQ3ODIxLWRjOTEtNDE3OC04NDdmLTQyYzk4ZmJjZDQwN1wvZGVvZ3F2OC1jNjE0MDQ5Yi1iY2Y2LTQ0MGEtOTc2NS05ZjBiODI5OWUzNTAucG5nIn1dXSwiYXVkIjpbInVybjpzZXJ2aWNlOmZpbGUuZG93bmxvYWQiXX0.asrcbJKBG2WDOG76pc9cFPMQIZZvMiVkV0Qz47u2_14) )

#### MUNCH frees your mind from the trouble of remembering things you need to do, so that you can focus on eating. It's
- Text-based
- Easy to learn
- **Beautiful** user interface

It's amazing :star_struck: :star_struck:

**All you have to do is,**
1. Download it
2. Click it
3. Add your tasks
4. Have faith in it :pray:
5. Have ur meal :hamburger:

#### MUNCH's features
- [x] Simple todo tasks
- [x] Managing deadlines 
- [ ] Reminders (coming soon)

#### If you are a Java programmer, you can use it to practice Java too! Here's the main method
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
#### So what are you waiting for? Join MUNCH and forget your worries :stuck_out_tongue_winking_eye:

Commit:
Add Gradle support

YZmunchmunch committed:
Level 1 changes

YZmunchmunch committed:
Level-2 Changes

YZmunchmunch committed:
A-Classes

YZmunchmunch committed:
Level-4

YZmunchmunch committed:
Level-3

YZmunchmunch committed:
Level-3

YZmunchmunch committed:
Level-5

YZmunchmunch committed:
Level-6

YZmunchmunch committed:
A-TextUiTesting

YZmunchmunch committed:
Level-7 (on a new branch)

YZmunchmunch committed:
Updated Level-7

YZmunchmunch committed:
Changes made to Level-7

YZmunchmunch committed:
branch with Level-8

Jarrett0203 commented on src/main/java/AddTasks/Deadlines.java:
```diff
@@ -0,0 +1,25 @@
+package AddTasks;
```
Perhaps you could change the package name to all lowercase? Similarly for the `Exceptions` class as well.

Jarrett0203 commented on src/main/java/Munch.java:
```diff
@@ -0,0 +1,151 @@
+        System.out.println("Wrong format for date! [Format: dd/MM/yyyy]");
+    }
+
+    public static LocalDate convert(String date) {
```
Maybe you can change the method name to `convertDate` to be more specific?

Jarrett0203 commented on src/main/java/AddTasks/Deadlines.java:
```diff
@@ -0,0 +1,25 @@
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
+public class Deadlines extends Task {
```
Maybe you can change the class name to a singular form? Similarly for the `Events` class as well.

Jarrett0203 commented on src/main/java/FileSave.java:
```diff
@@ -0,0 +1,52 @@
+            }
+            FileWriter munch = new FileWriter(path);
+            BufferedWriter munchWrite = new BufferedWriter(munch);
+            for(int i = 0; i < tasks.size(); i++) {
```
Maybe you can add a space between `for` and `(` ?

YZmunchmunch committed:
A-MoreOOP

YZmunchmunch committed:
A-Packages

YZmunchmunch committed:
Merge remote-tracking branch 'origin/add-gradle-support'

YZmunchmunch committed:
A-Gradle

YZmunchmunch committed:
A-JUnit

YZmunchmunch committed:
A-Jar

YZmunchmunch committed:
A-JavaDoc

YZmunchmunch committed:
A-JavaDoc

YZmunchmunch committed:
A-CodingStandard

YZmunchmunch committed:
Level-9

Infrix commented on src/main/java/munch/TaskList.java:
```diff
@@ -0,0 +1,142 @@
+
+public class TaskList {
+
+    public static void run(ArrayList<Task> tasks, String word, String[] words) throws IncompleteInputException, InvalidInputException {
```
Perhaps this method is a little too complicated to be in TaskList? Here, the TaskList is performing the job of parsing input, interpreting the command and running the command

An abstract command class could be created for different commands to inherit from, for them to handle their own execution

glyfy replied:
Perhaps create an abstract command class to reduce code?

Infrix commented on src/main/java/munch/TaskList.java:
```diff
@@ -0,0 +1,142 @@
+            tasks.add(todos);
+            Ui.addTaskMessage();
+            System.out.println(todos);
+            System.out.println("Now you have " + tasks.size() + " task(s) in the list ~~");
```
Should the job of printing be handled by the UI instead?

glyfy replied:
should printing be handled by UI class?

YZmunchmunch committed:
Update README.md

YZmunchmunch committed:
Level-10

YZmunchmunch committed:
Merge branch 'master' of https://github.com/YZmunchmunch/ip
Merge master

YZmunchmunch committed:
Certain parts of the code are not properly abstracted.

Making sure code follows good coding standard allows for better readability and oop.

Ensured that in a method, the level of abstractions between individual lines of codes are the same.

Pull out the behaviour when user inputs "list" or "find" into seperate methods, rather than computing it within the original getResponse method.

YZmunchmunch committed:
Merge pull request #3 from YZmunchmunch/branch-A-CodeQuality

Certain parts of the code are not properly abstracted.

YZmunchmunch committed:
Check for duplicates in new tasks being added.

Adding a checker method before each tasks are being added into the tasklist.

Doing so will improve abstraction, since the computation of checking duplicates will not be done in each adding tasks method.

YZmunchmunch committed:
Final changes to the chatbot, as well as cleaning up code

YZmunchmunch committed:
Added screenshot

YZmunchmunch committed:
changed to correct screenshot

YZmunchmunch committed:
Recommit for mistyped tag "BCD-Extension"

YZmunchmunch committed:
Update User Guide
